OpenRA.Game/Widgets/WorldInteractionControllerWidget.cs

```diff
@@ -226,13 +262,18 @@ public override bool HandleKeyPress(KeyInput e)
 
 		static IEnumerable<Actor> SelectActorsInBox(World world, int2 a, int2 b, Func<Actor, bool> cond)
 		{
-			return world.ScreenMap.ActorsInBox(a, b)
+			if (a == b || (a - b).Length > 50)
```

Comment:
This doesn't look right - you're mixing concerns.  Try moving this check up a level, or into its own method that gets called instead.